debug.py

```diff
@@ -0,0 +1,48 @@
+import cv2
+import numpy as np
+
+def _resize(img): #for debug purposes
+    width = 400
+    scale_factor = width / img.shape[1]
+    height = int(img.shape[0] * scale_factor)
+    dimension = (width, height)
+    im = cv2.resize(img, dimension, interpolation = cv2.INTER_AREA)
+    return im
+
+def debug_threshold(img) -> int:
+    '''
+    Function opens image with a threshold slider to adjust it in real time.
+    The inputed value effects the lower value [cv2.threshold(img, <VALUE>, 255, cv2.THRESH_BINARY)]
+    '''
+    # Callback function for the trackbar
+    def update_threshold(threshold_value):
+        _, thresholded_img = cv2.threshold(gray_img, threshold_value, 255, cv2.THRESH_BINARY)
+        cv2.imshow('Thresholded Image', _resize(thresholded_img))
+
+    gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+
+    # Create a window to display the image
+    cv2.namedWindow('Thresholded Image')
+
+    # Initialize threshold value
+    threshold_value = 120
+
+    # Create a trackbar
+    cv2.createTrackbar('Threshold', 'Thresholded Image', threshold_value, 255, update_threshold)
+
+    # Wait for ESC key to exit
+    while cv2.getWindowProperty('Thresholded Image', cv2.WND_PROP_VISIBLE) >= 1:
+        value = cv2.getTrackbarPos('Threshold', 'Thresholded Image')
+        key = cv2.waitKey(1) & 0xFF
+        if key == 27:  # ESC key
+            break
+
+    # Close all windows
+    cv2.destroyAllWindows()
+    return value
```

Comment:
@davide710 Why should it not work on black and white images? It converts colored ones to black and white ones aswell, so it shouldn't be a problem, right? 

The thing with solely black and white images is that the have a fixed value white(0,0,0) and black(255,255,255) and the function has therefore no effect on the image.

PS: correct me if I'm wrong or misunderstood the problem

Reply:
While colored images are arrays of shape (rows, columns, 3), black and white images (maybe I should have said  gray) are arrays of shape (rows, columns), since there is no need to store 3 values for each pixel. That's why calling `cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)` on a gray image results in an error.

Reply:
@davide710 Ok thanks for clarification I understand the problem now. I'm going to add a check to the function:
```python
gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY) if img.ndim == 3 else img
```
This will only work if there are **no other types of images** I currently not know about (new to cv2). It works by checking if the array has a dimention of 3 by using `img.ndim`, which would correspond to a colored image otherwise it is gray and we asign it to `gray_img`.

Reply:
Seems right, there are other types of images but they are unlikely to be used in this project